### DISCUSSION TITLE
Version 1.4.0

### DISCUSSION THREAD

**Discussion by MLWinther** [BASTAcode/BASTA#47] [Announcements]
Updated features, added minor features and extension of documentation.

Major updates:

- The installation of BASTA has been made easier, by making it pip installable. Users will still need to clone the repository, but the installation from there is essentially a one-liner. If updating from a previous version, remember to remove the environmnet variables (set in e.g. .bashrc), which are no longer needed.
- The frequency glitch fitting module have been updated to reflect the [Glitchpy](https://github.com/kuldeepv89/GlitchPy) routines. Helium glitch signatures can now be determined from observed frequencies and fitted separately or alongside frequency ratios with estimated correlations between them.
- A new section describing all input control options in BASTA has been added to the documentation, along with a list of all parameters known by BASTA.
- The routine to read and convert ascii files containing individual frequencies have been updated to automatically assign columns based on the file header and checking of values. The example using individual frequencies have been updated accordingly.

Minor changes:

- The `nottrusted` feature used to exclude individual frequency modes have been renamed to `excludemodes` for clarity.
- The Sobol routine have been updated to rely on the implementation by [Scipy](https://docs.scipy.org/doc/scipy/reference/generated/scipy.stats.qmc.Sobol.html), and does therefore not rely on the compilation of external Fortran routines. However, due to slight differences in the schemes, it is not backwards compatible with previous versions.

New minor features:

- If no radial order of individual frequencies are provided, they can be automatically estimated from division by the large frequency separation.
- The `onlyradial` input key can now be set to exclude all non-radial frequency modes from the fit.
- The `nameinplots` input key places the star identifier on the corner plot and Kiel diagram.
- The `dnubias` input can be provided to quadraticly add a bias to the error of the automatically computed large frequency separation, based on the radial individual frequency modes.

<hr /><em>This discussion was created from the release <a href='https://github.com/BASTAcode/BASTA/releases/tag/v1.4.0'>Version 1.4.0</a>.</em>

**Comment by MLWinther** [comment]
It should be noted that the new install of BASTA relies on features introduced in newer versions of pip (around version 23). 
Should the installation process fail with an error akin to "unable to find setup.py", this indicates that your version of pip needs to be updated, which can be done by
`pip install --upgrade pip`

**Comment by Mortal** [comment]
I would expect that you can install BASTA without cloning the repo, using the following line:        pip install https://github.com/BASTAcode/BASTA/archive/refs/heads/main.zipI'm not able to test it myself at the moment, but I thought I'd share it here.

**Reply by jakobmoss** [reply]
Indeed! Thanks for the tip. And for the development branch:
```
pip install https://github.com/BASTAcode/BASTA/archive/refs/heads/devel.zip
```
